### ISSUE TITLE
[Python] Implement new Concept Exercise: for loops

### ISSUE THREAD

**Issue by DavidGerva** [exercism/v3#1394]
This issue describes how to implement the `for loops` concept exercise for the Python track.

## Getting started

**Please please please read the docs before starting.** Posting PRs without reading these docs will be a lot more frustrating for you during the review cycle, and exhaust Exercism's maintainers' time. So, before diving into the implementation, please read up on the following documents:

- [The features of v3](https://github.com/exercism/v3/blob/master/docs/concept-exercises.md).
- [Rationale for v3](https://github.com/exercism/v3/blob/master/docs/rationale-for-v3.md).
- [What are concept exercise and how they are structured?](https://github.com/exercism/v3/blob/master/docs/features-of-v3.md)

Please also watch the following video:

- [The Anatomy of a Concept Exercise](https://www.youtube.com/watch?v=gkbBqd7hPrA).

## Goal

This concept exercise should convey a basic understanding of `for loops` in Python. The primary goal is to introduce the student to the usage of `for` loops with the `for ... in` pattern, one of the most used loop form in python.

## Learning objectives

- use a for loop to iterate over an iterable object with the `for ... in` pattern. The Membership testing concept (`in`) is required to address this concept.

## Out of scope

- `while` loops
- exiting loops shouldn't be a required knowledge
- async for loops (advanced)

## Concepts

- `for ... in` loops
- iteration, ideally without previous knowledge of the iteration concept

## Prerequisites

- membership testing `in`
- iterable objects base knowledge
- collections

## Resources to refer to

- [The for loop](https://yawpitchroll.com/posts/the-35-words-you-need-to-python/#for)

### Hints

- TBD?

### After

Nothing

## Representer

No changes required.

## Analyzer

No changes required.

## Implementing

Tests should be written using unittest.TestCase and the test file named bool_basic_test.py.

## Help

If you have any questions while implementing the exercise, please post the questions as comments in this issue.

## Edits
-

**Comment by aimorris** [comment]
Should be merged with #1432 and #1430

**Comment by DavidGerva** [comment]
Merged into #1430
